### PR TITLE
Significantly improve another exception that currently does not print any text.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by tjhei** [comment]
Printing the number of active cells seems to be more natural to me instead of the number of vertices. Is there any reason you went with vertices?

**Comment by bangerth** [comment]
I don't know -- that was what the assertion tested. In the end, what matters is that the exception printed some text. Whether the number is that of the vertices or that of the cells is immaterial -- the text says that the triangulation should be empty but isn't. I think the actual value printed is not important, just that it's nonzero.

(I'm not entirely sure whether the number of cells is relevant here -- the levels and vertices array we test may not hold any active cells, just unused slots, but even that is forbidden.)

(Addresses #610)

File: source/grid/tria.cc [hunk 1]
```diff
@@ -8932,9 +8932,9 @@ void
 Triangulation<dim, spacedim>::
 copy_triangulation (const Triangulation<dim, spacedim> &old_tria)
 {
-  Assert (vertices.size() == 0, ExcTriangulationNotEmpty());
-  Assert (levels.size () == 0, ExcTriangulationNotEmpty());
-  Assert (faces == NULL, ExcTriangulationNotEmpty());
+  Assert (vertices.size() == 0, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
+  Assert (levels.size () == 0, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
+  Assert (faces == NULL, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
```
Comment (tjhei) [inline source/grid/tria.cc]: would it make sense to change this to
```
Assert (vertices.size() == 0 && 
    levels.size () == 0 &&
    faces == NULL, 
    ExcTriangulationNotEmpty(vertices.size(), levels.size()));
```

File: source/grid/tria.cc [hunk 2]
```diff
@@ -9015,9 +9015,9 @@ create_triangulation (const std::vector<Point<spacedim> >    &v,
                       const std::vector<CellData<dim> > &cells,
                       const SubCellData &subcelldata)
 {
-  Assert (vertices.size() == 0, ExcTriangulationNotEmpty());
-  Assert (levels.size() == 0, ExcTriangulationNotEmpty());
-  Assert (faces == NULL, ExcTriangulationNotEmpty());
+  Assert (vertices.size() == 0, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
+  Assert (levels.size() == 0, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
+  Assert (faces == NULL, ExcTriangulationNotEmpty(vertices.size(), levels.size()));
```
Comment (tjhei) [inline source/grid/tria.cc]: same as above

**Comment by tjhei** [comment]
Please decide if you want to do the suggested change and/or just merge yourself.

**Comment by bangerth** [comment]
I followed your suggestion.

**Comment by tjhei** [comment]
Wait, what is going on here. You merged and then pushed another commit?

**Comment by bangerth** [comment]
github shows it wrongly. The additional commit was part of the merge and is now in master.

**Pull request merged**